packages/sources/nomics/test/integration/adapter.test.ts

```diff
@@ -0,0 +1,82 @@
+import { AdapterRequest } from '@chainlink/types'
+import request from 'supertest'
+import * as process from 'process'
+import { server as startServer } from '../../src'
+import { mockCryptoResponseSuccess, mockGlobalMarketResponseSuccess } from './fixtures'
+import * as nock from 'nock'
+import * as http from 'http'
+
+beforeAll(() => {
+  process.env.CACHE_ENABLED = 'false'
+  process.env.API_KEY = process.env.API_KEY || 'fake-api-key'
+  if (process.env.RECORD) {
+    nock.recorder.rec()
+  }
+})
+
+afterAll(() => {
+  if (process.env.RECORD) {
+    nock.recorder.play()
+  }
+
+  nock.restore()
+  nock.cleanAll()
+  nock.enableNetConnect()
+})
+
+describe('execute', () => {
+  const id = '1'
+  let server: http.Server
+  const req = request('localhost:8080')
+  beforeAll(async () => {
+    server = await startServer()
+  })
+  afterAll((done) => {
+    server.close(done)
+  })
```

Comment:
Could we do something like this instead? As long as there's only on 

```suggestion
describe('execute', () => {
  const id = '1'
  let server: http.Server
  const req = request('localhost:8080')
  
  beforeAll(async () => {
    process.env.CACHE_ENABLED = 'false'
    process.env.API_KEY = process.env.API_KEY || 'fake-api-key'
    if (process.env.RECORD) {
      nock.recorder.rec()
    }
    server = await startServer()
  })
  afterAll((done) => {
    if (process.env.RECORD) {
      nock.recorder.play()
    }

    nock.restore()
    nock.cleanAll()
    nock.enableNetConnect()
    server.close(done)
  })
```

Reply:
yeah, makes sense